File: 0881_boatsToSavePeople/main.cpp
#include <vector>
#include <algorithm>

class Solution {
public:
    int numRescueBoats(std::vector<int>& people, int limit) {
        std::sort(people.begin(), people.end());
        int l = 0;
        int r = people.size() - 1;
        int res = 0;
        while(l <= r) {
            if(people[l] + people[r] <= limit) {
                ++l;
            }
            --r;
            ++res;
        }

        return res;        
    }
};

File: 0881_boatsToSavePeople/main.py
from typing import List

class Solution:
    def numRescueBoats(self, people: List[int], limit: int) -> int:
        people.sort()
        l = 0
        r = len(people) - 1
        res = 0
        while l < r:
            if people[r] + people[l] <= limit:
                l += 1
            r -= 1
            res += 1
        return res